org.lflang.tests/src/org/lflang/tests/serialization/SerializationTest.java

```diff
@@ -0,0 +1,28 @@
+package org.lflang.tests.serialization;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+import org.lflang.Target;
+import org.lflang.tests.AbstractTest;
+import org.lflang.tests.Configurators;
+import org.lflang.tests.TestBase;
+import org.lflang.tests.TestBase.Message;
+import org.lflang.tests.TestBase.TestLevel;
+import org.lflang.tests.TestRegistry.TestCategory;
+
+public class SerializationTest extends TestBase {
+
+    protected SerializationTest() {
+        super(Target.ALL);        
+    }
+
```

Comment:
I'm not sure if I understand this comment.

Should we add a separate test for the single-threaded C runtime?
It shouldn't be important for the C serialization tests I think since I imagine all of them will eventually be federated.

Reply:
```suggestion
    @Test
    public void runSerializationTestsWithThreadingOff() {
        Assumptions.assumeTrue(supportsSingleThreadedExecution(), Message.NO_SINGLE_THREADED_SUPPORT);
        runTestsForTargets(Message.DESC_SERIALIZATION,
                TestCategory.SERIALIZATION::equals, Configurators::disableThreading,
                TestLevel.EXECUTION, false);
    }
}
```